admin/api.go

```diff
@@ -108,6 +123,33 @@ func (a *AdminAPI) GetState() (adminschema.State, error) {
 	return state, nil
 }
 
+type ClientRegistrationRequest struct {
+	IsAdmin bool                `json:"isAdmin"`
+	Client  oidc.ClientMetadata `json:"client"`
+}
+
+func (a *AdminAPI) CreateClient(req ClientRegistrationRequest) (oidc.ClientRegistrationResponse, error) {
+	if err := req.Client.Valid(); err != nil {
+		return oidc.ClientRegistrationResponse{}, mapError(err)
+	}
+	// metadata is guarenteed to have at least one redirect_uri by earlier validation.
+	id, err := oidc.GenClientID(req.Client.RedirectURIs[0].Host)
+	if err != nil {
+		return oidc.ClientRegistrationResponse{}, mapError(err)
+	}
+	c, err := a.clientIdentityRepo.New(id, req.Client)
+	if err != nil {
+		return oidc.ClientRegistrationResponse{}, mapError(err)
+	}
+	if req.IsAdmin {
+		// TODO(ericchiang): Put this in a transaction
+		if err := a.clientIdentityRepo.SetDexAdmin(c.ID, req.IsAdmin); err != nil {
```

Comment:
Will make `clientIdentityRepo.New` take a `is_admin` argument and do that there.